packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -163,13 +168,122 @@ class IosDomainError extends DomainError {
     'Ensure your domain is accessible without any redirects.',
   );
 
-  /// TODO(hangyujin): There are sub checkes of this check, add them and add links when finalized.
-  /// AASA file format follows guidelines.
-  static const fileFormat = IosDomainError(
-    'Apple-App-Site-Association file format is incorrect',
-    'This test checks that your Apple-App-Site-Association file '
-        'follows the correct format guidelines.',
-    'Ensure your Apple-App-Site-Association file follows the correct format guidelines.',
+  static IosDomainError iosFileFormatDomainError({
+    required List<AASAfileFormatSubCheck> subcheckErrors,
+  }) =>
+      IosDomainError(
+        'Apple-App-Site-Association file format is incorrect',
+        'This test checks that your Apple-App-Site-Association file '
+            'follows the correct format guidelines.',
+        'Ensure your Apple-App-Site-Association file follows the correct format guidelines.',
+        subcheckErrors: subcheckErrors,
+      );
+
+  final List<AASAfileFormatSubCheck> subcheckErrors;
+}
+
+class AASAfileFormatSubCheck extends CommonError {
+  const AASAfileFormatSubCheck(title, explanation)
+      : super(title, explanation, '');
+
+  static const appLinksFormat = AASAfileFormatSubCheck(
+    'Applinks format',
+    'This test checks if the `applinks` property holds an object.',
+  );
+  static const appLinksSubstitutionVariablesFormat = AASAfileFormatSubCheck(
+    'Applinks substitution variables format',
+    'This test checks if the `applinks.SubstitutionVariables` property has a valid '
+        'substitution variable format. Ref - '
+        'https://developer.apple.com/documentation/bundleresources/applinks/substitutionvariables',
+  );
+  static const defaultsFormat = AASAfileFormatSubCheck(
+    'Applinks defaults format',
+    'This test checks for `applinks.defaults`. Ref - '
```

Comment:
this seems incomplete. should this be "This test checks that the `applinks.defaults` property only holds the keys `caseSensitive` and `percentEncoded`. "?

then you can get rid of the sentence after the docs link too

Reply:
updated, thanks!